### PR TITLE
simplify docker-compose.yml

### PR DESCRIPTION
**Pull request by YANG-DB**
### Description
Simplify docker-compose to help user faster adoption and bootstrap

### Issues Resolved
https://github.com/opensearch-project/dashboards-observability/issues/762

### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -20,54 +19,37 @@ networks:
...
-      retries: 4
-    ports:
-      - "9200:9200"
-      - "9600:9600"
-
-  # OpenSearch store - node2
-  opensearch-node2:
-    image: opensearchproject/opensearch:${VERSION} # This should be the same image used for opensearch-node1 to avoid issues
-    container_name: opensearch-node2
+  opensearch:
```
Comment (joshuali925) [inline docker-compose.yml:22]: nice, i was wondering why we needed 2 nodes for this

File: docker-compose.yml [hunk 2]
```diff
@@ -20,54 +19,37 @@ networks:
...
+    image: opensearchstaging/opensearch:3.0.0
```
Comment (joshuali925) [inline docker-compose.yml:23]: the current staging image `opensearchstaging/opensearch:3.0.0` only contains 1 plugin
```
opensearch-security
```

users might need other plugins for observability playground to work correctly

File: docker-compose.yml [hunk 3]
```diff
@@ -20,54 +19,37 @@ networks:
...
-      - node.name=opensearch-node2
-      - discovery.seed_hosts=opensearch-node1,opensearch-node2
-      - cluster.initial_cluster_manager_nodes=opensearch-node1,opensearch-node2
+      - node.name=opensearch
+      - discovery.seed_hosts=opensearch
+      - cluster.initial_cluster_manager_nodes=opensearch
       - bootstrap.memory_lock=true
       - "OPENSEARCH_JAVA_OPTS=-Xms512m -Xmx512m"
+      - "DISABLE_INSTALL_DEMO_CONFIG=true"
+      - "DISABLE_SECURITY_PLUGIN=true"
```
Comment (joshuali925) [inline docker-compose.yml:33]: why disable security plugin?

**Review by joshuali925 — COMMENTED** [review]
you might also want to limit the max log size for ~data-prepper and~ (i forgot this one doesn't have data-prepper) opensearch, i saw them growing fast and takes a lot of disk space

File: docker-compose.yml (hunk 3 above)
Comment (YANG-DB) [reply docker-compose.yml:33]: it only for local tests and for working for development internally

File: docker-compose.yml (hunk 2 above)
Comment (YANG-DB) [reply docker-compose.yml:23]: I'll add documentation for this...

**Review by joshuali925 — APPROVED** [review]

**Comment by joshuali925** [comment]
i think there's one more place that's outdated
https://github.com/opensearch-project/dashboards-observability/blob/0d7c96734020f9a7985963af587e47e1ff5debf7/Using-Docker.md?plain=1#L44-L46

File: .env [hunk 4]
```diff
@@ -1,5 +1,3 @@
 # version for opensearch & opensearch-dashboards docker image
-VERSION=2.6.0
-# auth details - change according to your system
-USER=admin
-PASSWORD=admin
+VERSION=2.9.0
```
Comment (pjfitzgibbons) [inline .env:2]: Will this cause an issue with building on 'main', b/c the OS 3.0 docker image published as "latest" (there is no docker-hub published tag for 3.0)

**Review by pjfitzgibbons — APPROVED** [review]

File: .env (hunk 4 above)
Comment (YANG-DB) [reply .env:2]: this is subject to the one using this docker - he needs to change this for his own needs - Its also stated inside the docker usage readme

**Pull request merged**